src/inmanta/server/protocol.py

```diff
@@ -208,24 +209,34 @@ async def stop(self) -> None:
 
         order = list(reversed(self._get_slice_sequence()))
 
-        exceptions: dict[str, Exception] = {}
+        pre_stop_exceptions: dict[str, Exception] = {}
+        stop_exceptions: dict[str, Exception] = {}
+
         for endpoint in order:
             try:
                 LOGGER.debug("Pre Stopping %s", endpoint.name)
                 await endpoint.prestop()
             except Exception as e:
-                exceptions[endpoint.name] = e
+                pre_stop_exceptions[endpoint.name] = e
+
+        if pre_stop_exceptions:
+            LOGGER.debug("The following errors occurred during pre-stopping: %s", str(pre_stop_exceptions))
 
-        LOGGER.debug("The following errors occurred during pre-stopping: %s", str(exceptions))
-        exceptions.clear()
         for endpoint in order:
             try:
                 LOGGER.debug("Stopping %s", endpoint.name)
                 await endpoint.stop()
             except Exception as e:
-                exceptions[endpoint.name] = e
-
-        LOGGER.debug("The following errors occurred during stopping: %s", str(exceptions))
+                stop_exceptions[endpoint.name] = e
+        if stop_exceptions:
+            LOGGER.debug("The following errors occurred during stopping: %s", str(stop_exceptions))
+
+        if pre_stop_exceptions or stop_exceptions:
+            raise ExceptionGroup(
```

Comment:
Shouldn't we use `BaseExceptionGroup` instead so that we don't get a TypeError if one of the exeption is a `BaseException` but not an `Exception`.